### DISCUSSION TITLE
[Need help] Disappearing colon due to *Indent after a colon

### DISCUSSION THREAD

**Discussion by ullaskunder3** [microsoft/vscode-cpptools#11854] [Q&A] [answered]
I am experiencing an issue in the C++ language mode where, when I enter a colon : and then press Enter for a new line, the colon disappears it will work if I use space after colon . This behavior is unexpected

using: C/C++ v1.18.5 Microsoft Extension

- if i think it because of this `Indent after a colon (Global User Snippet)`

https://github.com/microsoft/vscode-cpptools/assets/66258652/97131beb-594f-48fb-aa0f-aba8e68b527a

**Comment by bobbrow** [comment]
Hi @ullaskunder3,

Did you add that global snippet, or do you have an extension installed that added it?  The C++ extension does not add that snippet so this issue appears to be specific to how your instance of VS Code is configured.

**Reply by ullaskunder3** [reply]
NO, this is it
 
```json

{
  "C_Cpp.default.intelliSenseMode": "gcc-x64",
  "window.titleBarStyle": "custom",
  "window.title": "New Day New Hope",
  "editor.fontFamily": "'Cascadia Code', Consolas, 'Courier New', monospace",
  "terminal.integrated.fontFamily": "Cascadia Code PL",
  "editor.fontLigatures": "'calt', 'ss01'",
  "editor.fontSize": 15,
  "editor.tokenColorCustomizations": {
    "textMateRules": [
      {
        "scope": [
          "comment",
          "keyword",
          "storage.modifier",
          "storage.type.class.js",
          "storage.type.js",
          "variable.language.this.js",
          "variable.parameter.js",
          "storage.type.function.js",
          "variable.parameter.ts",
          "keyword.operator.new.ts",
          // C++
          "storage.type.built-in.primitive.cpp"
        ],
        "settings": {
          "fontStyle": "italic",
        }
      },
      {
        "scope": [
          "keyword.control.conditional",
          "keyword.operator",
          // C++
          "storage.type.namespace.directive.cpp"
        ],
        "settings": {
          "fontStyle": ""
        }
      }
    ]
  },
  "C_Cpp.intelliSenseCacheSize": 0,
  "window.commandCenter": false,
  "[javascript]": {
    "editor.defaultFormatter": "esbenp.prettier-vscode"
  },
  "[typescriptreact]": {
    "editor.defaultFormatter": "vscode.typescript-language-features"
  },
  "editor.quickSuggestions": {
    "strings": true
  },
  "[xml]": {
    "editor.defaultFormatter": "DotJoshJohnson.xml"
  },
  "redhat.telemetry.enabled": false,
  "[jsonc]": {
    "editor.defaultFormatter": "vscode.json-language-features"
  },
  "C_Cpp.clang_format_fallbackStyle": "Google",
  "[typescript]": {
    "editor.defaultFormatter": "esbenp.prettier-vscode"
  },
  "terminal.integrated.enableMultiLinePasteWarning": false,
  "liveServer.settings.donotShowInfoMsg": true,
  "editor.lineHeight": 1.4,
  "[json]": {
    "editor.defaultFormatter": "vscode.json-language-features"
  },
  "[javascriptreact]": {
    "editor.defaultFormatter": "vscode.typescript-language-features"
  },
  "workbench.iconTheme": "material-icon-theme",
  "terminal.integrated.minimumContrastRatio": 7,
  "editor.minimap.enabled": false,
  "workbench.colorTheme": "Zeus Turquoise",
  "quokka.automaticRestart": true,
  "quokka.compactMessageOutput": true,
  "quokka.automaticStartRegex": "true",
  "wallaby.startAutomatically": true,
  "rsp-ui.enableStartServerOnActivation": [
    {
      "id": "redhat.vscode-community-server-connector",
      "name": "Community Server Connector",
      "startOnActivation": true
    }
  ],
  "javascript.updateImportsOnFileMove.enabled": "always",
  "workbench.sideBar.location": "right",
  "audioCues.taskCompleted": "on",
  "editor.bracketPairColorization.independentColorPoolPerBracketType": true,
  "editor.bracketPairColorization.enabled": true,
  "editor.guides.bracketPairs": true,
}
```

**Reply by bobbrow** [reply, marked as answer]
Those are your settings, but what we need to look for is your global user snippets.  Try running the "Snippets: Configure User Snippets" command in VS Code.  It should show you whether you have a global snippets file for your installation or any of the languages already.  Pick one of those files and look for the snippet called "Indent after a colon".

![image](https://github.com/microsoft/vscode-cpptools/assets/12818240/98a5d16c-632b-4855-9fa6-f8e3a53b47ad)

I suspect this is related to Python, do you have any Python extensions installed?  If this advice doesn't help you locate the issue, you might want to try asking some folks over in the Python extension.

**Reply by ullaskunder3** [reply]
thanks :+1:  got it,...

I removed this and worked fine thanks

```
	"Indent After Colon": {
		"prefix": ":",
		"body": [
			"  $0"
		],
		"description": "Indent after a colon"
	}	
```

**Comment by ullaskunder3** [comment]
@bobbrow  how to configure new line indentation after a colon for C++ example

```cpp
class Apple {
public:
  // Cursor here after pressing Enter, I want the indentation on the next line to be aligned with the colon above.
}
```

**Reply by bobbrow** [reply]
You might have more success using the `editor.formatOnType` setting than trying to write your own rules.  Though I don't know of a formatter that aligns code underneath the colon like that unless you set your tab width to 6 or 7. That's an uncommon tab width and may not be something you'd want to get used to since you are not likely to see other projects formatted that way (my opinion).

If you want to get into the details of language configurations without using a formatter, there's a complex set of rules that VS Code supports and you'd have to write a full language configuration to use them.  https://code.visualstudio.com/api/language-extensions/language-configuration-guide (see "Indentation Rules").  I don't necessarily recommend going this route and even we as an extension don't try to define indentation rules in this way - we rely on formatters to do the job instead since they have more context.
